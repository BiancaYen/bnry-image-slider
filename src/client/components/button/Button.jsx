// Button

import React from 'react';
import PropTypes from 'prop-types';

// Constants
import alignmentConstants from '../constants';

// Styles
import { Container, Tooltip } from './styles';

// Prop Types
const propTypes = {
    alignment: PropTypes.oneOf([
        alignmentConstants.left,
        alignmentConstants.right
    ]).isRequired,
    children: PropTypes.string.isRequired,
    tooltip: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired
};

const Button = ({
    alignment,
    children,
    tooltip,
    onClick
}) => {
    return (
        <Container>
            <Tooltip alignment={alignment}>
                {tooltip}
            </Tooltip>
            <button
                type="type"
                onClick={event => onClick(event.preventDefault())}
            >
                {children}
            </button>
        </Container>
    );
};

Button.propTypes = propTypes;
Button.alignment = alignmentConstants;

export default Button;
