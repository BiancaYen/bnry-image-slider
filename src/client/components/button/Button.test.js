// Button Test

import { mount, shallow } from 'enzyme';
import React from 'react';
import sinon from 'sinon';

// Components
import Button from './Button';

// Constants
const alignment = 'left';
const title = 'Title';
const tooltip = 'Tooltip';

describe('Button', () => {
    it('renders with all required prop types', () => {
        const component = shallow(
            <Button
                alignment={alignment}
                tooltip={tooltip}
                onClick={() => {}}
            >
                {title}
            </Button>
        );

        expect(component).toMatchSnapshot();
    });

    it('asserts that prop types are correct', () => {
        const onClick = sinon.spy();
        const component = mount(
            <Button
                alignment={alignment}
                tooltip={tooltip}
                onClick={onClick}
            >
                {title}
            </Button>
        );

        expect(component.find('button').text()).toEqual(title);
        expect(component.find('div div').text()).toEqual(tooltip);

        component.find('button').simulate('click');
        expect(onClick).toHaveProperty('callCount', 1);
    });
});
