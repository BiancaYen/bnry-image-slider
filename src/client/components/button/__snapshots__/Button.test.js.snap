// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders with all required prop types 1`] = `
<Container>
  <Tooltip
    alignment="left"
  >
    Tooltip
  </Tooltip>
  <button
    onClick={[Function]}
    type="type"
  >
    Title
  </button>
</Container>
`;
